frappe/desk/doctype/workspace/workspace.py

```diff
@@ -251,6 +249,12 @@ def new_page(new_page):
 	):
 		frappe.throw(_("Cannot create private workspace of other users"), frappe.PermissionError)
 
+	elif not frappe.has_permission(doctype="Workspace", ptype="create"):
+		frappe.flags.error_message = _("User {0} does not have the permission to create a Workspace.").format(
+			frappe.bold(frappe.session.user)
+		)
+		raise frappe.PermissionError
+
```

Comment:
```suggestion
```

Just remove `ignore_permissions=True` few line below.

Reply:
Yeah i had tried that, but ig because of how the error message is defined in `has_permission`, this error is thrown for Workspace permissions - 

<img width="612" alt="Screenshot 2024-05-31 at 12 28 38 PM" src="https://github.com/frappe/frappe/assets/40693548/e4a618c9-f6f0-43ca-8bbd-3683dea1b47c">

------

https://github.com/frappe/frappe/blob/1269e3117048daac20d0c03dc6486c143867b0fe/frappe/permissions.py#L133-L140

Since for ptype create, the document name doesn't exist yet, there should be a separate message without the docname but wasn't sure if changing it directly here would be okay so that's why handled it like that.